experimental/ModelBuilder/ModelRunner.cpp

```diff
@@ -50,8 +50,10 @@ namespace llvm {
 extern Pass* createLowerMatrixIntrinsicsPass();
 }  // end namespace llvm
 
-void mlir::ModelRunner::compile(CompilationOptions compilationOptions,
-                                llvm::ArrayRef<const std::string> runtime) {
+void mlir::ModelRunner::compile(
+    CompilationOptions compilationOptions,
+    llvm::ArrayRef<const std::string> runtime,
+    llvm::ArrayRef<std::pair<std::string, void*>> extra_symbols) {
```

Comment:
If you were using the same callback mechanism that is used by the engine, you wouldn't force the client to materialize the array with this exact `pair<std::string, void*>`.

Reply:
Ack. I think having a callback-taking-callback here, encoding the array and pairs as a form of continuation-passing style, is making the API too complex, and this is probably fine for simple uses...